sdk/src/main/java/ly/count/android/sdk/ModuleRequestQueue.java

```diff
@@ -217,6 +232,56 @@ synchronized public void requestQueueEraseAppKeysRequestsInternal() {
         attemptToSendStoredRequestsInternal();
     }
 
+    /**
+     * Send request data after removing the predefined keys
+     */
+    synchronized public void addDirectRequestInternal(@NonNull Map<String, String> requestMap) {
+        L.i("[Countly] Calling addDirectRequest");
+        if (!_cly.isInitialized()) {
+            L.e("Countly.sharedInstance().init must be called before adding direct request, returning");
+            return;
+        }
+
+        if (!consentProvider.anyConsentGiven()) {
+            L.e("[ModuleRequestQueue] addDirectRequest, no consent is given, returning");
+            return;
+        }
+
+        if (requestMap == null || requestMap.isEmpty()) {
+            L.e("[ModuleRequestQueue] addDirectRequest, provided requestMap was null or empty, returning");
+            return;
+        }
+
+        // Filtering and removing predefined/restricted keys
+        Map<String, String> filteredRequestMap = new HashMap<>();
+        for (Map.Entry<String, String> entry : requestMap.entrySet()) {
+            String key = entry.getKey();
+            Object value = entry.getValue();
+            boolean isPreDefinedKey = false;
+
+            for (String preDefinedKey : preDefinedKeys) {
+                if (preDefinedKey.equals(key)) {
+                    //if it's a predefined field
+                    isPreDefinedKey = true;
+                    L.w("[ModuleRequestQueue] addDirectRequest, removing provided key: [" + key + "] is a restricted key.");
+                    break;
+                }
+            }
+
+            if (!isPreDefinedKey) {
+                filteredRequestMap.put(key, value.toString());
+            }
+        }
+
+        int requestDataCount = requestMap.size();
+        int filteredDataCount = filteredRequestMap.size();
+        int delta = requestDataCount - filteredDataCount;
+        if(delta > 0) {
+            L.w("[ModuleRequestQueue] addDirectRequest, [" + delta + "] restricted keys are removed");
+        }
+        requestQueueProvider.sendDirectRequest(filteredRequestMap);
```

Comment:
you need to double check that there are any entries remaining. Print error  if that is the case